File: lib/qt5/LocationListComboBox.cpp
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <pkmn/qt5/LocationListComboBox.hpp>

#include <pkmn/database/lists.hpp>

namespace pkmn { namespace qt5 {

    LocationListComboBox::LocationListComboBox(
        const QString &game,
        bool wholeGeneration,
        QWidget* parent
    ): QComboBox(parent)
    {
        std::vector<std::string> locations = pkmn::database::get_location_list(
                                                 game.toStdString(),
                                                 wholeGeneration
                                             );
        for(auto iter = locations.begin(); iter != locations.end(); ++iter) {
            addItem(QString::fromStdString(*iter));
        }

        setEditable(false);
    }

}}

// Generated at build time
#include "../include/pkmn/qt5/moc_LocationListComboBox.cpp"

File: lib/swig/java/ComboBoxes/LocationListComboBox.java
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.JComboBox;

/**
 * A JComboBox filled with a list of locations in the given game (or generation).
 */
public class LocationListComboBox extends JComboBox<String> {
    public LocationListComboBox(String game, boolean wholeGeneration) {
        StringVector locationList = Database.getLocationList(game, wholeGeneration);
        for(long i = 0; i < locationList.size(); i++) {
            this.addItem(locationList.get((int)i));
        }
        this.setEditable(false);
    }
}
